docs/apis/subsystems/communication/index.md

```diff
@@ -0,0 +1,244 @@
+                avatar: $courseimage ?: null,
+                instance: $course,
+            );
+```
+
+#### Delete a room
+
+$communication->delete_room() method is used to add an ad-hoc to delete a room in the communication provider. The associated task for this also removes all the members from
+the room before deleting the room. This is destructive and might remove all the messages and other data associated with the room. Usage of this should be done with caution.
+
+#### Add members to a room
+
+$communication->add_members_to_room() method is used to add an ad-hoc to add members to a room in the communication provider. The user id of each user to be added to the provider room is also stored in
+the communication_user table for user mapping with the provider.
+
+```php
+public function add_members_to_room(
+        array $userids,
+        bool $queue = true,
+    )
+```
+
+This method accepts the following parameters:
+
+- An array of user ids to be added to the provider room.
+- A boolean value to indicate if the task should be added to the queue or run immediately. This is false by default. This is added to ensure the cases where the mapping should be created but the task might not be needed at this point.
+
+```php
+// First initialize the instance, provider is not required here, it will initialize the enabled instance.
+$communication = \core_communication\api::load_by_instance(
+            context: $context,
+            component: 'core_course',
+            instancetype: 'coursecommunication',
+            instanceid: $course->id,
+        );
+
+// Now add the members.
+$communication->add_members_to_room(
+                userids: $enrolledusers,
+            );
+```
+
+#### Update the membership of a room
+
+$communication->update_room_membership() method is used to add an ad-hoc to update the membership of a room in the communication provider. The user mapping for each of the users are also reset to allow task to update the users from the task.
+
+```php
+public function update_room_membership(
+        array $userids,
+        bool $queue = true,
+    )
+```
+
+This method accepts the same parameters as the add_members_to_room() method and the usage is also the same.
+
+#### Remove members from a room
+
+$communication->remove_members_from_room() method is used to add an ad-hoc to remove members from a room in the communication provider. The users are flagged as deleted in the communication_user table to allow the task to remove the users from the provider room.
+
+```php
+public function remove_members_from_room(
+        array $userids,
+        bool $queue = true
+    )
+```
+
+This method accepts the same parameters as the add_members_to_room() method and the usage is also the same.
+
+#### Show the communication room creation status notification
+
+$communication->show_communication_room_status_notification() is a special method to have a UI element to show the status of the room creation. If the room is ready, it will
```

Comment:
```suggestion
`$communication->show_communication_room_status_notification()` is a special method to have a UI element to show the status of the room creation. If the room is ready, it will
```